modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs

```diff
@@ -7,22 +7,56 @@
 using System.Collections.Generic;
 using System.Runtime.CompilerServices;
 
-// TODO: Add comments describing what this class does. It is not obvious.
-
 namespace Godot
 {
+    /// <summary>
+    /// Godot's global functions.
+    /// </summary>
     public static partial class GD
     {
+        /// <summary>
+        /// Decodes a byte array back to a <c>Variant</c> value.
+        /// If <paramref name="allowObjects"/> is <see langword="true"/> decoding objects is allowed.
+        ///
+        /// WARNING: Deserialized object can contain code which gets executed.
+        /// Do not set <paramref name="allowObjects"/> to <see langword="true"/>
+        /// if the serialized object comes from untrusted sources to avoid
+        /// potential security threats (remote code execution).
+        /// </summary>
+        /// <param name="bytes">Byte array that will be decoded to a <c>Variant</c>.</param>
+        /// <param name="allowObjects">If objects should be decoded.</param>
+        /// <returns>The decoded <c>Variant</c>.</returns>
         public static object Bytes2Var(byte[] bytes, bool allowObjects = false)
         {
             return godot_icall_GD_bytes2var(bytes, allowObjects);
         }
 
+        /// <summary>
+        /// Converts from a <c>Variant</c> type to another in the best way possible.
+        /// The <paramref name="type"/> parameter uses the <see cref="Variant.Type"/> values.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// var a = new Vector2(1, 0);
+        /// // Prints 1
+        /// GD.Print(a.Length());
+        /// var b = GD.Convert(a, Variant.Type.String)
+        /// // Prints 6 as "(1, 0)" is 6 characters
+        /// GD.Print(b.Length);
+        /// </code>
+        /// </example>
+        /// <returns>The <c>Variant</c> converted to the given <paramref name="type"/>.</returns>
         public static object Convert(object what, Variant.Type type)
         {
             return godot_icall_GD_convert(what, type);
         }
 
+        /// <summary>
+        /// Converts from decibels to linear energy (audio).
+        /// </summary>
+        /// <seealso cref="Linear2Db(real_t)"/>
+        /// <param name="db">Decibels to convert.</param>
+        /// <returns>Volume as linear energy.</returns>
```

Comment:
This statement isn't quite clear enough on its own, since volume can also mean 3D space. It's likely fine because of the context, but I think we should make it clearer:

```suggestion
        /// <returns>Audio volume as linear energy.</returns>
```